docs/quick-start.md

```diff
@@ -171,6 +205,44 @@ Here, we combined the [`flatMap`](programming-guide.html#transformations), [`map
 {% endhighlight %}
 
 </div>
+<div data-lang="r" markdown="1">
+
+{% highlight r %}
+> reduce(map(textFile, function(line) { length(strsplit(line, " ")[[1]]) }), function(a, b) {max(a, b)})
+[1] 14
+{% endhighlight %}
+
+This first maps a line to an integer value, creating a new RDD. `reduce` is called on that RDD to find the largest line count. The arguments to `map` and `reduce` are R [anonymous functions](http://adv-r.had.co.nz/Functional-programming.html#anonymous-functions),
+but we can also pass any top-level R function we want.
+For example, we'll define a `mymax` function to make this code easier to understand:
+
+{% highlight r %}
+> mymax <- function(a, b) { max(a, b) }
+> reduce(map(textFile, function(line) { length(strsplit(line, " ")[[1]]) }), mymax)
+[1] 14
+{% endhighlight %}
+
+One common data flow pattern is MapReduce, as popularized by Hadoop. Spark can implement MapReduce flows easily:
+
+{% highlight r %}
+> wordCounts <- reduceByKey(map(flatMap(textFile, function(line) strsplit(line, " ")[[1]]), function(word) list(word, 1)), "+", 2)
```

Comment:
Can you split this into multiple lines, so it becomes more readable ? i.e. something like
```
wordCounts <- reduceByKey(
  map(
    flatMap(textFile, function(line) strsplit(line, " ")[[1]]), 
    function(word) list(word, 1)), 
 "+",  2)
```